src/leases/components/leaseSections/contract/Contract.js

```diff
@@ -27,40 +27,75 @@ const Contract = ({contract}: Props) => {
-                  <p className='text-no-margin' key={get(pledge_book, 'pledge_book_number', '')}>{get(pledge_book, 'pledge_book_number', '')}, {helpers.formatDate(get(pledge_book, 'pledge_book_date', ''))}</p>
-                )}
-              </div>
             </Column>
             <Column medium={4}>
               <label>Allekirjoituspäivämäärä</label>
               <p>{contract.signing_date ? helpers.formatDate(get(contract, 'signing_date', '–')) : '–'}</p>
-
-              <label>Vuokravakuusnumero</label>
-              <p>{contract.lease_deposit_number ? get(contract, 'lease_deposit_number', '–') : '–'}</p>
-
+            </Column>
+            <Column medium={4}>
+              <label>Kommentti allekirjoitukselle</label>
+              <p>{contract.signing_date_comment ? get(contract, 'signing_date_comment', '–') : '–'}</p>
+            </Column>
+          </Row>
+          <Row>
+            <Column medium={4}>
+              <label>Järjestelypäätös</label>
+              <p>{contract.setup_decision ? get(contract, 'setup_decision', '–') : '–'}</p>
+            </Column>
+            <Column medium={4}>
+              <label>Päätös</label>
+              {/* link to chosen rule */}
+              <p>–</p>
+            </Column>
+            <Column medium={4}>
               <label>KTJ dokumentti</label>
               {/* link to KTJ */}
               <p>Vuokraoikeustodistus</p>
             </Column>
+          </Row>
+          <Row>
+            <Column medium={4}>
+              <label>Vuokravakuusnumero</label>
+              <p>{contract.lease_deposit_number ? get(contract, 'lease_deposit_number', '–') : '–'}</p>
+            </Column>
             <Column medium={4}>
-              <label>Laitostunnus</label>
-              <p>{contract.administration_number ? get(contract, 'administration_number', '–') : '–'}</p>
-
               <label>Vuokravakuus alku- ja loppupäivämäärä</label>
               <p>{helpers.formatDate(get(contract, 'lease_deposit_starting_date', '–'))} – {helpers.formatDate(get(contract, 'lease_deposit_ending_date', '–'))}</p>
-
-              <label>Päätös</label>
-              {/* link to chosen rule */}
-              <p>–</p>
+            </Column>
+            <Column medium={4}>
+              <label>Kommentti</label>
+              <p>{contract.lease_deposit_comment ? get(contract, 'lease_deposit_comment', '–') : '–'}</p>
             </Column>
           </Row>
+          <Row>
+            <Column medium={4}>
+              <label>Laitostunnus</label>
+              <p>{contract.administration_number ? get(contract, 'administration_number', '–') : '–'}</p>
+            </Column>
+          </Row>
+          {contract.pledge_books.length > 0 &&
+          <Row>
+            <Column>
+              <p className='green-subtitle'>Panttikirjat</p>
+            </Column>
+          </Row>
+          }
+          {contract.pledge_books.length > 0 && contract.pledge_books.map((pledge_book, index) =>
```

Comment:
Page will crash if contract.pledge_books is null. I would be good to check is contract.pledge_books defined
e.g.
{contract.pledge_books && contract.pledge_books.length > 0 && contract.pledge_books.map((pledge_book, index) =>

Reply:
of course! forgot that :/